src/System.Devices.Gpio/GPIORef.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Devices.Gpio
+{
+    /// <summary>
+    /// GPIO - Basic Scenarios
+    /// 
+    /// Most GPIO implementations share a core set of functionality to allow basic on/off control
+    /// of pins. At the very least, we should support these.
+    /// - Open a object that is a representation of a pin with the given pin number
+    /// - Support closing a pin to release the resources owned by that pin object
+    /// - Represent the mode of the pin that details how the pin handles reads and writes (e.g. Input, Output)
+    /// - Allow a resistor to be added to a pin such that it can be set as pullup or pulldown (or no pull)
+    /// - Support setting a PWM value on a pin
+    /// </summary>
+    #region GPIO - Basic Scenarios
+
+    public partial class GPIOPin : IDisposable
```

Comment:
This does not follow our casing guidelines. It should be GpioPin. Also, I would remove "partial" from API spec. It's a source code concept.